### DISCUSSION TITLE
Arduino MEGA + ESP8266 WIFI Inbuilt Board

### DISCUSSION THREAD

**Discussion by supekshala** [JAndrassy/WiFiEspAT#48] [General]
I recently started a project to read analog values from arduino mega and send those values to the REST API endpoint using POST.
To do this I brought an Arduino MEGA + ESP8266 WIFI Inbuilt Board board. I think I really messed up with firmware. I can upload sketches to mega but I can't connect to the ESP module. (I have tried with the DIP switch). Could you help me to find the right firmware and fix this? That would a huge help.
Thank You!

**Comment by JAndrassy** [comment]
the Mega+WiFi is the ideal board for this library.
switch the switches to connect CH340 to esp8266 (5 and 6 ON and 7 too for esp8266 flashing mode) and upload AT firmware. see the [README](https://github.com/jandrassy/WiFiEspAT/blob/master/README.md) for AT firmware versions and upload commands.
then switch CH340 to ATmega and ATmega to esp8266 (1 to 4 ON, 5,6,7 off) and the Serial3 switch to Serial3.
in WiFiEspAT example sketches then always change Serial1 to Serial3.

**Reply by supekshala** [reply]
Thank you! I upgrade the firmware but still AT commands not working. I got this
![eee1](https://user-images.githubusercontent.com/14821352/121825587-ef995780-ccd0-11eb-929b-54cc56090f8a.JPG)
 Firmware installation 
![firmwa](https://user-images.githubusercontent.com/14821352/121825616-16f02480-ccd1-11eb-96d1-f6f4907364dc.JPG)
![ssa](https://user-images.githubusercontent.com/14821352/121825635-30916c00-ccd1-11eb-8066-ff204c573994.JPG)

**Reply by JAndrassy** [reply]
as I write in the [README](https://github.com/jandrassy/WiFiEspAT/blob/master/README.md) for AT 1.7.4 use:
esptool.py write_flash --flash_size 2MB-c1 0x0 boot_v1.7.bin 0x01000 at/1024+1024/user1.2048.new.5.bin 0x1fb000 blank.bin 0x1fc000 esp_init_data_default_v08.bin 0xfe000 blank.bin 0x1fe000 blank.bin

**Reply by supekshala** [reply]
THANK YOU it worked!! 
![ggggggga](https://user-images.githubusercontent.com/14821352/121982482-30669e80-cdad-11eb-9005-2c8fe60a98fc.JPG)

![hhhhh](https://user-images.githubusercontent.com/14821352/121982394-0e6d1c00-cdad-11eb-8a51-a962a23c87d5.JPG)

**Reply by Bordman-ger** [reply]
Hi, after spending hours to get this done on my device I want to add a hint: I am Windows user and tried to get the Flash_download_tool.EXE running- with lot of errors and hints on other sides (new version 3.9.3 is faulty, Memsize can not be set, factory mode to be used etc....) I finaly gave up und used the Python Tool... and it worked like a charm! 
So do not waste time with the Windows Exe... it seems not to work any longer...
